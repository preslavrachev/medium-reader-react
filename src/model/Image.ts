export default class Image {
  private _imageId: string;

  public get imageId() : string {
    return this._imageId;
  }
  
}