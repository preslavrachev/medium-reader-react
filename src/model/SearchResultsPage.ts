import ArticleBasicInfo from 'ArticleBasicInfo';

class Payload {
  private _value: PayloadValue;

  public get value() : PayloadValue {
    return this._value;
  }
  
}

class PayloadValue {
  private _posts: Array<ArticleBasicInfo>;

  public get posts(): Array<ArticleBasicInfo> {
    return this._posts;
  }
}

/**
 * SearchResultsPage
 */
export default class SearchResultsPage {
  private _success: boolean;
  private _payload: Payload;

  public get success(): boolean {
    return this._success;
  }

  public get payload(): Payload {
    return this._payload;
  }
}