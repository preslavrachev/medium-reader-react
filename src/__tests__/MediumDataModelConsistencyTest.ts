import SearchResultsPage from '../model/SearchResultsPage';
import Image from '../model/Image';
import fixture from './fixtures/mock-medium-list-response.json';

it('asserts that some subtitltes are present', () => {
  const mockData: SearchResultsPage = fixture as SearchResultsPage;
  const posts = mockData.payload.value.posts;
  const subtitles:string[] = posts
    .map(p => p.virtuals.subtitle)
    .filter(p => p !== undefined);
  
  expect(subtitles.length).toEqual(posts.length);
  expect(subtitles).toMatchSnapshot();
})

it('asserts that the images are present', () => {
  const mockData: SearchResultsPage = fixture as SearchResultsPage;
  const images:string[] = mockData.payload.value.posts
    .map(p => p.virtuals.previewImage ? p.virtuals.previewImage.imageId : 'imageIdPlaceholder.png');

  expect(images).toMatchSnapshot();
});
