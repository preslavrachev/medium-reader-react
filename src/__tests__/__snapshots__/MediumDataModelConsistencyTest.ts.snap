// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`asserts that some subtitltes are present 1`] = `
Array [
  "This is the subtitle for post #1",
  "This is the subtitle for post #2",
  "This is the subtitle for post #3",
]
`;

exports[`asserts that the images are present 1`] = `
Array [
  "imageId1.jpg",
  "imageId2.jpg",
  "imageIdPlaceholder.png",
]
`;
